main.go

```diff
@@ -0,0 +1,297 @@
+					}).Fatal("Hierarchy directory not found")
+				} else {
+					log.WithFields(log.Fields{
+						"path": includePath,
+					}).Info("Ignoring missing hierarchy directory")
+				}
+			}
+		}
+
+		// Break the for loop on error including EOF
+		if err != nil {
+			break
+		}
+	}
+	if err != io.EOF {
+		checkForError(err)
+	}
+	return hierarchy
+}
+
+// mergeYamls walks through all the folders in the hierarchy
+// and merges all files matching the pattern into the structure
+// overwriting any existing values
+// and finally exports it as a new YAML file
+func mergeYamls(hierarchy []string, fileFilter string, outputFile string) {
+	// Initialize variables
+	var data map[string]interface{}
+	counter := 0
+
+	for _, includePath := range hierarchy {
+		log.WithFields(log.Fields{
+			"path": includePath,
+		}).Debug("Inspecting folder")
+
+		// Merge in every file matching the pattern
+		for _, file := range getFiles(includePath, fileFilter) {
+			// Generate an old version of YAML for comparison
+			oldYaml, err := yaml.Marshal(&data)
+			checkForError(err)
+
+			// Import the next file
+			log.WithFields(log.Fields{
+				"path": file,
+			}).Info("Importing file")
+			mergeFile, err := ioutil.ReadFile(file)
+			checkForError(err)
+			mergeData := make(map[string]interface{})
+			err = yaml.Unmarshal([]byte(mergeFile), &mergeData)
+			checkForError(err)
+
+			err = mergo.Merge(&data, mergeData, mergo.WithOverride)
+			checkForError(err)
+
+			// Generate the new YAML and print the unified diff to the trace output
+			newYaml, err := yaml.Marshal(&data)
+			checkForError(err)
+			log.Trace(diff.Diff(string(oldYaml), string(newYaml)))
+
+			counter++
+		}
+	}
+
+	log.WithFields(log.Fields{
+		"count": counter,
+	}).Info("Completed merging all files")
+
+	// Writing the output file
+	log.WithFields(log.Fields{
+		"path": outputFile,
+	}).Info("Writing output file")
+	yamlDoc, err := yaml.Marshal(&data)
+	err = ioutil.WriteFile(outputFile, yamlDoc, 0660)
+
+	checkForError(err)
+}
+
+// getFiles lists all files in a given path and returns a list of matching the fileFilter
+func getFiles(includePath string, fileFilter string) []string {
+	var includeFiles []string
+	files, err := ioutil.ReadDir(includePath)
+	checkForError(err)
+	for _, fileInfo := range files {
+		if !fileInfo.IsDir() {
+			filePath := path.Join(includePath, fileInfo.Name())
+			r, err := regexp.MatchString(fileFilter, fileInfo.Name())
+			if err == nil && r {
+				includeFiles = append(includeFiles, filePath)
+				log.WithFields(log.Fields{
+					"file": filePath,
+				}).Debug("Adding file to list")
+			} else {
+				log.WithFields(log.Fields{
+					"file": filePath,
+				}).Debug("Ignoring file")
+			}
+		}
+	}
+
+	return includeFiles
+}
+
+// ReplaceEnvironmentVariables replaces all variable names in a string with the content defined on the OS
+// If a variable is not defined, it will fail to avoid any unintended results
+func ReplaceEnvironmentVariables(str string) string {
```

Comment:
Do you need it to be exportable? Other functions start with a lowerCase. There's multiple references that need to be updated.
```suggestion
func replaceEnvironmentVariables(str string) string {
```